### PR TITLE
Add seed data

### PR DESCRIPTION
**Pull request by townerr**
Added seed data creation to context constructor
removes previous seed data and remakes the data

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Looks good.

**Review by mattmarnien — DISMISSED** [review]

File: FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs [hunk 1]
```diff
@@ -37,5 +51,100 @@ public IMongoCollection<Notification> Notifications
...
+                noteCol.InsertOne(note1);
+                noteCol.InsertOne(note2);
+                noteCol.InsertOne(note3);
+                noteCol.InsertOne(note4);
+
+                return true;
+            }
+            catch
+            {
+                return false;
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs:156]: either we should have logging here, or this catch should be narrowed way down to only the expected exception

Comment (escalonn) [reply FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs:156]: this is not how we log exceptions. this way (implicitly calling ToString) throws out most of the information about the exception, including the stack trace.. all you need to do is inject an `ILogger<NotificationsContext>`.

**Review by escalonn — COMMENTED** [review]
https://github.com/2011-fakebook-project3/notifications/pull/42#discussion_r556866765

**Review by SukafuDesu — DISMISSED** [review]

File: FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs [hunk 2]
```diff
@@ -78,16 +84,17 @@ public bool ClearNotifications(IMongoCollection<User> userCol, IMongoCollection<
 
                 return true;
             }
-            catch(Exception ex)
+            catch
             {
-                Debug.WriteLine("Could not clear previous seed data" + ex);
+                _logger.LogWarning("Unable to clear seed data");
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs]: half the point of logging here is to log the exception, use the overload that includes the exception with the string message

File: FakebookNotifications/FakebookNotifications.Testing/Repo Tests/NotificationRepoTests.cs [hunk 3]
```diff
@@ -8,16 +8,21 @@
...
 namespace FakebookNotifications.Testing
 {
     public class NotificationRepoTests
     {
         private Mock<IOptions<NotificationsDatabaseSettings>> _mockSettings;
         private NotificationsDatabaseSettings settings;
+        private readonly ILogger<NotificationsContext> _logger;
 
-        public NotificationRepoTests()
+        public NotificationRepoTests(ILogger<NotificationsContext> logger)
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.Testing/Repo Tests/NotificationRepoTests.cs]: there isn't a dependency injector in xunit. just mock the logger, or instantiate a `NullLogger<NotificationsContext>`.

File: FakebookNotifications/FakebookNotifications.WebApi/Program.cs [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
         public static void Main(string[] args)
         {
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder
+                    .AddFilter("Microsoft", LogLevel.Information)
+                    .AddFilter("System", LogLevel.Information)
+                    .AddConsole();
+            });
+            ILogger logger = loggerFactory.CreateLogger<Program>();
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.WebApi/Program.cs]: what are we doing here

File: FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs [hunk 5]
```diff
@@ -84,9 +84,9 @@ public bool ClearNotifications(IMongoCollection<User> userCol, IMongoCollection<
 
                 return true;
             }
-            catch
+            catch(Exception ex)
             {
-                _logger.LogWarning("Unable to clear seed data");
+                _logger.LogWarning("Unable to clear seed data" + ex);
```
Comment (escalonn) [inline FakebookNotifications/FakebookNotifications.DataAccess/NotificationsContext.cs]: again, that's not it. this calls ToString which throws away most of the details of the exception.

there's overloads for every `Log*` method where the first parameter is a whole exception object.

**Review by escalonn — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**